python/taichi/lang/kernel_arguments.py

```diff
@@ -61,8 +61,9 @@ def decl_any_arr_arg(dtype, dim, element_shape, layout):
     if layout == Layout.AOS:
         element_dim = -element_dim
     return AnyArray(
-        _ti_core.make_external_tensor_expr(dtype, dim, arg_id, element_dim),
-        element_shape, layout)
+        _ti_core.make_external_tensor_expr(dtype, dim, arg_id, element_dim,
```

Comment:
btw we wanted to pass in a TensorType instead of a flattened primitive dtype but that can be a followup I think.

Reply:
Let's make it in a followup PR, shall we leave a tracker?